### ISSUE TITLE
Add `unwatch` method

### ISSUE THREAD

**Issue by cheap-glitch** [fregante/fit-textarea#19]
This method would:
 - drop the cached mirror (and the window listeners if the cache is empty)
 - drop the listener on the textarea element
 - maybe drop the listener on the form too? (but other textareas might still need it)

**Comment by fregante** [comment]
`{signal}`? 🙂

**Comment by fregante** [comment]
Keep in mind that RG is stuck on ~~v3~~ _v2_ until further notice. I basically have to rewrite this module or just release v2 as v4. As you saw in the `min-height` thread, there's no winning here. 

I published the module hoping that it would be a straightforward solution compared to the rest, but Safari absolutely hates the double resizing on every keystroke, so… unless there's a solution to that, I almost feel like I should retire the module or keep v2 with a warning for long pages + Safari.

**Comment by cheap-glitch** [comment]
> Keep in mind that RG is stuck on v3

You mean v2, right? I'm using v3 in another extension and it works fine on Chrome & Firefox, but I do see spikes in CPU usage when typing (like going from 3% to ~18% on Chromium).

In that case why not keep the best of both worlds with the help of a little `userAgent` magic?

**Comment by fregante** [comment]
Right. v2 was slow on Safari and v3 is slow on Chrome, which is why I reverted to v2 and disabled the feature in Safari.


> In that case why not keep the best of both worlds 

![5b0821d415e9f917c2730963](https://user-images.githubusercontent.com/1402241/180398809-ab79b9d1-85f1-42bb-b418-5e0daa54c877.jpg)

**Comment by fregante** [comment]
The problem is that neither implementation is good, as seen by the resize issue. Plus the box-sizing issue.

I'd probably rather detect the "fit" speed automatically and dynamically ~~memoize~~ _debounce_ accordingly. e.g. if it takes 100ms, _debounce_ to 500ms. If it takes 5ms, leave it raw on `input`

Edit: in v2, since it's simpler.
